semgrep-core/src/runner/Run_semgrep.ml

```diff
@@ -511,11 +541,12 @@ let semgrep_with_rules config ((rules, invalid_rules), rules_parse_time) =
            in
 
            let xtarget = xtarget_of_file config xlang file in
-           let match_hook str env matched_tokens _opt_taint_finding =
+           let match_hook str match_ =
              if config.output_format = Text then
-               let xs = Lazy.force matched_tokens in
-               print_match ~str config.match_format config.mvars env
-                 Metavariable.ii_of_mval xs
+               let xs = Lazy.force match_.Pattern_match.tokens in
+               print_match ~str config.match_format config.mvars
+                 match_.Pattern_match.env Metavariable.ii_of_mval xs
```

Comment:
at this point maybe we should just pass the whole match_ to print_match and let print_match internally
access its different elements.
We could also pass the whole config instead of passing those 2 config arguments..